src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -1054,6 +1090,43 @@ public static SAMFileWriter createCommonSAMWriterFromFactory(
     }
 
     /**
+     * Create a common SAMFileWriter from a factory for use with GATK tools. Assumes that if the factory has been set
+     * to create an index, the header must be set to coordinate sorted.
+     *
+     * @param outputPath if this file has a .cram extension then a reference is required. Can not be null.
+     * @param referenceFile the reference source to use. Can not be null if a output file has a .cram extension.
+     * @param header header to be used for the output writer
+     * @param preSorted if true then records must already be sorted to match the header sort order
+     * @param factory SAMFileWriterFactory factory to use
+     * @param openOptions (optional) NIO options specifying how to open the file
+     * @return SAMFileWriter
+     */
+    public static SAMFileWriter createCommonSAMWriterFromFactory(
+        final SAMFileWriterFactory factory,
+        final Path outputPath,
+        final File referenceFile,
+        final SAMFileHeader header,
+        final boolean preSorted,
+        OpenOption... openOptions)
+    {
+        Utils.nonNull(outputPath);
+        Utils.nonNull(header);
+
+        if (outputPath.toString().endsWith(CramIO.CRAM_FILE_EXTENSION)) {
+            if (null == referenceFile ) {
+                throw new UserException("A reference file is required for writing CRAM files");
+            }
+            // htsjdk doesn't currently support presorted & outputStream at the same time for CRAM :-(
```

Comment:
Not sure what this refers to - but see my comment below.

Reply:
This explains why we are not forwarding the "preSorted" flag. Because there is no overload that will take it.